File: Patrones Comportamiento/Template Method/TemplateMethod.py
# Template Method
class Process:
    def run(self):
        self.step1()
        self.step2()

class ConcreteProcess(Process):
    def step1(self):
        print("Paso 1")
    def step2(self):
        print("Paso 2")

ConcreteProcess().run()

File: Patrones Comportamiento/Template Method/TemplateMethod.js
// Template Method
class Process {
  run() {
    this.step1();
    this.step2();
  }
}

class ConcreteProcess extends Process {
  step1() {
    console.log("Paso 1");
  }
  step2() {
    console.log("Paso 2");
  }
}

new ConcreteProcess().run();
